packages/discord.js-utilities/src/lib/MessagePrompter.ts

```diff
@@ -0,0 +1,410 @@
+ * ```typescript
+ * const handler = new MessagePrompter('Are you sure you want to continue?');
+ * const confirmed = await handler.run(channel, author);
+ * ```
+ *
+ * @example
+ * ```typescript
+ * const handler = new MessagePrompter('Choose a number between 5 and 10?', {
+ * 		type: 'number',
+ * 		reactions: {
+ * 			start: 5,
+ * 			end: 10
+ * 		}
+ * });
+ * const confirmed = await handler.run(channel, author);
+ * ```
+ *
+ * @example
+ * ```typescript
+ * const handler = new MessagePrompter('Are you happy or sad?', {
+ * 		type: 'reaction',
+ * 		reactions: ['🙂', '🙁']
+ * });
+ * const confirmed = await handler.run(channel, author);
+ * ```
+ *
+ * @example
+ * ```typescript
+ * const handler = new MessagePrompter('Do you love me?', 'message');
+ * const confirmed = await handler.run(channel, author);
+ * ```
+ */
+export class MessagePrompter {
+	/**
+	 * The message that will been sent in [[MessagePrompter.run]]
+	 */
+	public message: IMessagePrompterMessage;
+
+	/**
+	 * The message that has been sent in [[MessagePrompter.run]]
+	 */
+	public appliedMessage: Message | null = null;
+
+	/**
+	 * The used options for this instance of [[MessagePrompter]]
+	 */
+	public options: IMessagePrompterOptions;
+
+	/**
+	 * The default options used if there is no type supplied.
+	 */
+	private defaultOptions: IMessagePrompterOptions = {
+		type: 'confirm',
+		reactions: {
+			confirm: '🇾',
+			cancel: '🇳'
+		},
+		timeout: 20 * 1000,
+		explicitReturn: false
+	};
+
+	/**
+	 * Constructor for the [[MessagePrompter]] class
+	 * @param message The message to send.
+	 * @param options Overrideable options if needed.
+	 */
+	public constructor(message: IMessagePrompterMessage, options?: Partial<IMessagePrompterOptions> | IMessagePrompterOptionsType) {
+		this.message = message;
+		const type: IMessagePrompterOptionsType = typeof options === 'string' ? options : options?.type ?? 'confirm';
+
+		this.options = {
+			...this.defaultOptions,
+			...(MessagePrompter.defaultOptions.get(type) ?? {}),
+			...((typeof options !== 'string' && options) ?? {})
+		};
+	}
+
+	/**
+	 * This executes the [[MessagePrompter]] and sends the message.
+	 * @param channel The channel to use.
+	 * @param authorOrFilter The author to validate.
+	 */
+	public run(channel: TextChannel | NewsChannel | DMChannel, authorOrFilter: User | CollectorFilter): Promise<IMessagePrompterReturn> {
+		switch (this.options.type) {
+			case 'confirm':
+				return this.runConfirm(channel, authorOrFilter);
+			case 'number':
+				return this.runNumber(channel, authorOrFilter);
+			case 'reaction':
+				return this.runReaction(channel, authorOrFilter);
+			case 'message':
+				return this.runMessage(channel, authorOrFilter);
+		}
+	}
+
+	/**
+	 * This executes the [[MessagePrompter]] and sends the message if [[IMessagePrompterOptions.type]] equals confirm.
+	 * The handler will wait for one (1) reaction.
+	 * @param channel The channel to use.
+	 * @param authorOrFilter The author to validate.
```

Comment:
```suggestion
	 * @param authorOrFilter An author object to validate or a [CollectorFilter](https://discord.js.org/#/docs/main/stable/typedef/CollectorFilter) predicate callback.
```
Just documenting the new feature (and follow suit for other functions)